app/services/force/event_subscriber_translate_service.rb

```diff
@@ -110,12 +110,30 @@ def translate_and_cache(event)
       )
     end
 
-    def translate_event_values(values)
-      logger("Translating values: #{values.inspect}")
+    def process_event_values(listing_id, values)
       return [] unless values&.values
 
+      listing = Request.new(parse_response: true).cached_get(
```

Comment:
Can we place the code for handling the diff in another function? 

We shouldn't have to get the listing everytime in this function, only if we encounter a diff object

Reply:
It's tough because we only know if it's a diff when we look at the individual values. Having to check if there are diffs ahead of processing means we have to loop through all the values an extra time. I pushed an update so the listing gets called on the first diff only. Let me know what you think!